src/main/java/org/testng/IDataProviderMethod.java

```diff
@@ -0,0 +1,41 @@
+package org.testng;
+
+import java.lang.reflect.Method;
+import java.util.List;
+
+/**
+ * Represents the attributes of a {@link org.testng.annotations.DataProvider} annotated method.
+ */
+public interface IDataProviderMethod {
```

Comment:
Why not providing `IDataProviderAnnotation getAnnotation()` instead of many getters?

Reply:
Right! :( It could be a next improvement :p